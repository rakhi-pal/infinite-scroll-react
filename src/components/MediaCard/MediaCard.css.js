import { makeStyles } from '@material-ui/core/styles';
const useStyles = makeStyles({
    card: {
      flex: '1 1 0',
      minWidth: 250,
      maxWidth: 250,
      minHeight: 270,
      borderRadius: 10,
      padding: 5,
      margin: '1rem',
      '&:hover': {
              transform: 'scale(1.3)'
      }
    },
    media: {
      height: 128,
      width: 128,
      borderRadius: 10
    },
  });

export {useStyles};