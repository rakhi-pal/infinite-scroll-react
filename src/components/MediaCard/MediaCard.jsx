import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import Card from '@material-ui/core/Card';
import CardMedia from '@material-ui/core/CardMedia';
import CardContent from '@material-ui/core/CardContent';
import Typography from '@material-ui/core/Typography';

const useStyles = makeStyles({
  card: {
    flex: '1 1 0',
    minWidth: 250,
    maxWidth: 250,
    minHeight: 270,
    borderRadius: 10,
    padding: 5,
    margin: '1rem',
    '&:hover': {
			transform: 'scale(1.3)'
    }
  },
  media: {
    height: 140,
  },
});

const MediaCard = React.forwardRef(({image, title, user}, ref) => {
  const classes = useStyles();

  return (
    <Card ref={ref} className={classes.card}>
      <CardMedia
        className={classes.media}
        image={user.picture.large}
        title={user.name.first +" " + user.name.last}
      />
      <CardContent>
          <Typography gutterBottom variant="h5" component="h2">
            {user.name.first +" " + user.name.last}
          </Typography>
          <Typography variant="body2" color="textSecondary" component="p">
            <label>{user.email}</label><br/>
            <label>phone: {user.phone}</label><br/>
            <label>cell: {user.cell}</label>
          </Typography>
        </CardContent>
    </Card>
  );
});

export default MediaCard;
