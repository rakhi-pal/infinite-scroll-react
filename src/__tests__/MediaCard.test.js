import { render } from '@testing-library/react';
import MediaCard from '../components/MediaCard/MediaCard';
import {create, act} from 'react-test-renderer'; 
import Card from '@material-ui/core';

test('renders MediaCard component', () => {
    const props = {
        user: {
            picture: {
                large: 'abc.png',
            },
            name: {
                first: 'first',
                last: 'last'
            },
            email: 'abc@test.com',
            phone: '1234565',
            cell: '1234567890'
        }
    }
    render(<MediaCard {...props} />);
});

test('Card component renders inside MediaCard component', () => {
    const props = {
        user: {
            picture: {
                large: 'abc.png',
            },
            name: {
                first: 'first',
                last: 'last'
            },
            email: 'abc@test.com',
            phone: '1234565',
            cell: '1234567890'
        }
    }

    let MediaCardComponent;
    act(() => {
        MediaCardComponent = create(<MediaCard {...props} />);
    })
    expect(MediaCardComponent.toJSON()).toMatchSnapshot();
});
