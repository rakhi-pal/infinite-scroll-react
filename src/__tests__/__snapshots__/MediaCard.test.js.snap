// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card component renders inside MediaCard component 1`] = `
<div
  className="MuiPaper-root MuiCard-root makeStyles-card-3 MuiPaper-elevation1 MuiPaper-rounded"
>
  <div
    className="MuiCardMedia-root makeStyles-media-4"
    style={
      Object {
        "backgroundImage": "url(\\"abc.png\\")",
      }
    }
    title="first last"
  />
  <div
    className="MuiCardContent-root"
  >
    <h2
      className="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom"
    >
      first last
    </h2>
    <p
      className="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary"
    >
      <label>
        abc@test.com
      </label>
      <br />
      <label>
        phone: 
        1234565
      </label>
      <br />
      <label>
        cell: 
        1234567890
      </label>
    </p>
  </div>
</div>
`;
